frontend/src/app/shared/components/storages/location-picker-modal/location-picker-modal.component.ts

```diff
@@ -170,4 +171,19 @@ export class LocationPickerModalComponent extends FilePickerBaseModalComponent {
 
     return this.text.tooltip.file_not_selectable;
   }
+
+  public newFolderDialog() {
+    const value = window.prompt(this.text.buttons.newFolder);
+    if (!value) { return; }
+
+    this.storageFilesResourceService.createFolder(
+      `${this.storage._links.self.href}/folders`,
```

Comment:
if you want to keep this URL building out of the modal, you can add it to the `locals` map. The `createFolderHref` is static, hence this could be an easy solution and you won't rebuilt it:

```typescript
    const locals = {
      projectFolderHref: this.projectStorage._links.projectFolder?.href,
      projectFolderMode: this.projectStorage.projectFolderMode,
      storage,
    };
```